torch/_dynamo/optimizations/torchxla_integration.py

```diff
@@ -73,12 +99,11 @@ def import_torchxla():
     import torch_xla.debug.metrics as metrics
 
 
-def extract_compiled_graph(model: torch.fx.GraphModule, example_inputs):
+def extract_compiled_graph(xla_model: torch.fx.GraphModule, xla_args):
     import_torchxla()
-    orig_device = example_inputs[0].device
-    xla_dev = xm.xla_device()
-    xla_model = copy.deepcopy(model).to(device=xla_dev)
-    xla_args = [arg.to(device=xla_dev) for arg in example_inputs]
+
+    # This call is critical to make sure xla_args' tensor id show up in graph_input_tensor_ids
+    xm.mark_step()
```

Comment:
Would be interesting to use `torch_xla._XLAC. _xla_tensors_report(0, str(xla_device))` to figure out what's the pending graph here. I think for the case that dynamo can not capture the whole graph, the pending graph is those model that fallback to lazy device. If user manually runs parts of the model on xla directly, this will also collapse those execution which I think is the correct behavior we want.

Reply:
@JackCaoG  The symptom is the model inputs does not show up as xla graph inputs. On the other hand I actually don't see pending graph from the API you mentioned.

Here is the command for you reference (repro with the Linear model):
```
GPU_NUM_DEVICES=1 python benchmarks/dynamo/torchbench.py --randomize-input --performance --trace-on-xla --backend=torchxla_trace_once --only path:/pytorch/myscripts/model_collection.py,class:LinearModel
```

Output:
```
Tensor: id=1, shape=f32[10,10]{1,0}, device=GPU:0, ir_nodes=1
IR {
  %0 = f32[10,10]{1,0} xla::device_data(), device=GPU:0, ROOT=0
}


Tensor: id=2, shape=f32[10]{0}, device=GPU:0, ir_nodes=1
IR {
  %0 = f32[10]{0} xla::device_data(), device=GPU:0, ROOT=0
}


Tensor: id=3, shape=f32[10,10]{1,0}, device=GPU:0, ir_nodes=1
IR {
  %0 = f32[10,10]{1,0} xla::device_data(), device=GPU:0, ROOT=0
}


Tensor: id=4, shape=f32[10]{0}, device=GPU:0, ir_nodes=1
IR {
  %0 = f32[10]{0} xla::device_data(), device=GPU:0, ROOT=0
}


Tensor: id=5, shape=f32[2,10]{1,0}, device=GPU:0, ir_nodes=1
IR {
  %0 = f32[2,10]{1,0} xla::device_data(), device=GPU:0, ROOT=0
}


Tensor: id=6, shape=f32[10,10]{1,0}, device=GPU:0, ir_nodes=1
IR {
  %0 = f32[10,10]{1,0} xla::device_data(), device=GPU:0, ROOT=0
}


Tensor: id=7, shape=f32[10]{0}, device=GPU:0, ir_nodes=1
IR {
  %0 = f32[10]{0} xla::device_data(), device=GPU:0, ROOT=0
}


Tensor: id=8, shape=f32[10,10]{1,0}, device=GPU:0, ir_nodes=1
IR {
  %0 = f32[10,10]{1,0} xla::device_data(), device=GPU:0, ROOT=0
}


Tensor: id=9, shape=f32[10]{0}, device=GPU:0, ir_nodes=1
IR {
  %0 = f32[10]{0} xla::device_data(), device=GPU:0, ROOT=0
}


Tensor: id=10, shape=f32[2,10]{1,0}, device=GPU:0, ir_nodes=1
IR {
  %0 = f32[2,10]{1,0} xla::device_data(), device=GPU:0, ROOT=0
}
```

Reply:
hmm OK, doesn't seem like we have any pending computation. I think we can ignore this for now since `mark_step` is a good workaround. We can revisit later.